inconspiquous/dialects/qu.py

```diff
@@ -51,6 +67,37 @@ def num_qubits(self) -> int:
         return 1
 
 
+@irdl_op_definition
+class FromBitsOp(IRDLOperation):
+    name = "qu.from_bits"
+    qubits = var_operand_def(BitType)
+    reg = result_def(QReg)
+
+
+@irdl_op_definition
+class ToBitsOp(IRDLOperation):
+    name = "qu.to_bits"
+    reg = operand_def(QReg)
+    qubits = var_result_def(BitType)
+
+
+@irdl_op_definition
+class CombineOp(IRDLOperation):
+    name = "qu.combine"
+    reg1 = operand_def(QReg)
+    reg2 = operand_def(QReg)
+    res = result_def(QReg)
+
+
+@irdl_op_definition
+class SplitOp(IRDLOperation):
+    name = "qu.split"
+    reg = operand_def(QReg)
+    split_index = prop_def(IntegerAttr)
```

Comment:
The `split_index` property shouldn't be needed here, as we should be able to tell where the split occurs by the result types.